### ISSUE TITLE
End of line comment in inventory files break [group:children] sections

### ISSUE THREAD

**Issue by ddelemeny** [ansible/ansible#8239]
##### Issue Type:

Bug Report
##### Ansible Version:

1.6.7
##### Environment:

N/A
##### Summary:

When a [group:children] type section ends with a comment, ansible doesn't match declared hosts hosts.
##### Steps To Reproduce:

inventory

```
[foo]
localhost 
[bar:children] # If this comment exists, hosts of group "bar" will not be matched
foo
```

test.yml

```
- hosts: bar
  user: ddelemeny
  tasks:
    - name: dbg
      debug:
```

```
ansible-playbook test.yml -i inventory
```
##### Expected Results:

Without the comment :

```
PLAY [bar] ******************************************************************** 

GATHERING FACTS *************************************************************** 
ok: [localhost]

TASK: [dbg] ******************************************************************* 
ok: [localhost] => {
    "msg": "Hello world!"
}

PLAY RECAP ******************************************************************** 
localhost                  : ok=2    changed=0    unreachable=0    failed=0   

```
##### Actual Results:

With the comment :

```
PLAY [bar] ******************************************************************** 
skipping: no hosts matched

PLAY RECAP ******************************************************************** 

```

**Comment by ddelemeny** [comment]
Haha, P5 indeed.

I've looked at the inventory parser and I think I could try to fix this one myself, and perhaps improve and refactor a bit where it's needed, but since it's a core element of ansible, I'd like to have some guidance on how to proceed.

Anyone ?

**Comment by Spedge** [comment]
Hey Damien. I'm new to the Ansible project and looking for something easy to tackle to get my hands dirty, this seems like a good place to start. Going to take a look and get back to you.

**Comment by Spedge** [comment]
Hi Damien, 

It looks as if comments after groups were fixed in v1.4.0 within /lib/ansible/inventory/ini.py (https://github.com/ansible/ansible/commit/7ecb5fbc9c4084ad2251acdeaa4bced21db22512) but because the ini parsing is done in several different places within the file, the same bug exists for not only after group:children but also variables too (according to the code). 

Going to set up some unit tests and then give it a try.

**Comment by guhcampos** [comment]
Hi, this looks a bit old but I can confirm the bug on 1.9.2

**Comment by jimi-c** [comment]
Hi all, sorry for the late follow-up on this. It appears this issue has been resolved as far back as 2.0.0.2 (that's the furthest back I went). As such, we'll go ahead and close this now.

If you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
